p2p/kademlia.py

```diff
@@ -320,16 +325,64 @@ def binary_get_bucket_for_node(buckets: List[KBucket], node: Node) -> KBucket:
         raise ValueError("No bucket found for node with id {}".format(node.id))
 
 
+class CallbackLock:
+    def __init__(self,
+                 callback: Callable,
+                 timeout: int=300) -> None:
```

Comment:
How about making the default here something like `2 * k_request_timeout` as by then we can be sure nobody should be waiting for the callback anyway?